### ISSUE TITLE
Add cmap.to_mpl() toplevel helper?

### ISSUE THREAD

**Issue by anntzer** [pyapp-kit/cmap#79]
* cmap version: 0.4
* Python version: 3.13
* Operating System: any

### Description

A toplevel helper `cmap.to_mpl("colormap:name")` would be quite practical, I believe.  (I realize it's close to a one-liner `def to_mpl(...): return cmap.Colormap(...).to_mpl(...)` although if you want to be as feature-complete as possible, to_mpl() could split the kwargs passed in between those that need to be passed to the Colormap constructor and those that need to be pass to the Colormap.to_mpl method.)

For completeness, one could also have `cmap.to_altair` and so on.

An alternate name would be `as_mpl` (`as_altair`, etc.).

Thoughts?

### What I Did

N/A

**Comment by tlambert03** [comment]
Hey @anntzer, I'm not opposed to it.

Is the ultimate goal just to make it easier to use cmap colormaps in matplotlib with `to_mpl('name')` instead of `Colormap('name').to_mpl()`?  

If so, another possibility we could look into is a function that you could call that would register all of the cmap namespace colormaps with matplotlib, such that you could just use the string names directly in matplotlib `cmap=` arguments.  would that be preferable?

**Comment by anntzer** [comment]
> Is the ultimate goal just to make it easier to use cmap colormaps in matplotlib with `to_mpl('name')` instead of `Colormap('name').to_mpl()`?

Yes.

> If so, another possibility we could look into is a function that you could call that would register all of the cmap namespace colormaps with matplotlib, such that you could just use the string names directly in matplotlib `cmap=` arguments. would that be preferable?

That would also work, I guess.

**Comment by apcamargo** [comment]
Does `cmap` provides functionality to register a given colormap easily? As far as I know, `Colormap('name').to_matplotlib()` and `Colormap('name').to_mpl()` don't register the colormap at all.

**Comment by tlambert03** [comment]
No, `cmap` doesn't currently provide any convenience functions around registering string names with matplotlib.  That would (I believe) be part of what might be done here.

**Comment by apcamargo** [comment]
I think adding a method to register a specific colormap could be useful. However, I'm not in favor of registering all of them at once. That's just a personal preference, though. If the registered colormaps have a unique prefix, I don't see any issues.

**Comment by tlambert03** [comment]
yeah, I agree.  it's too many.  I'll go ahead and add the top level `to_mpl` that @anntzer was requesting for now.  and will look into a convenience method for registering by name
